front/src/modules/simulationResult/components/SpaceTimeChart/useGetProjectedTrainOperationalPoints.ts

```diff
@@ -40,13 +59,44 @@ const useGetProjectedTrainOperationalPoints = (
         };
         const { operational_points } = await postPathProperties(pathPropertiesParams).unwrap();
 
-        setOperationalPoints(
-          operational_points as NonNullable<PathProperties['operational_points']>
-        );
+        let operationalPointsWithAllWaypoints = operational_points as NonNullable<
+          PathProperties['operational_points']
+        >;
+        // Check if there are vias added by map click and insert them in the operational points
+        if (trainScheduleUsedForProjection.path.some((step) => 'track' in step)) {
+          trainScheduleUsedForProjection.path.forEach((step, i) => {
+            if ('track' in step) {
+              const positionOnPath = pathfindingResult.path_item_positions[i];
+              const indexToInsert = operationalPointsWithAllWaypoints.findIndex(
+                (op) => op.position >= positionOnPath
+              );
+
+              const formattedStep: NonNullable<PathProperties['operational_points']>[number] = {
+                id: step.id,
+                extensions: {
+                  identifier: {
+                    name: t('requestedPoint', { count: indexToInsert }),
```

Comment:
`indexToInsert` is a bit weird to display to the user IMHO, since:

- It's 0-based, so can result in "requested point 0", which is scary for non-programmers.
- It's not the index in the list of waypoints requested by the user, it's an index in the path properties ranges? Not exactly sure.

Maybe `i + 1` would be a tad more informative. But maybe "point on track XXX" would be even better?

Reply:
Actually I discussed with @thibautsailly about that and he doesn't want to display the index among all ops but only from the "waypoint added by map click" list